pgcode: use XC instead of CDB

otan opened the pull request:
Release note (sql change): Change the pgerror code XC instead of CD
for CockroachDB specific errors. This is because the "C" class is
reserved for the SQL standard. The pgcode `CDB00` used for
unsatisfiable bounded staleness is now `XCUBS`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/70347)
<!-- Reviewable:end -->

knz reviewed (commented):
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/70347)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rafiss)

___
*[pkg/sql/pgwire/pgcode/codes.go, line 383 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/70347#-Mjo2BRnEbnQnrZQXTxb:-Mjo2BRoCYQwZuIWpxqJ:b-n0rrnk) ([raw file](https://github.com/cockroachdb/cockroach/blob/59f84d2290bfcff5a39574ead151445355a134fc/pkg/sql/pgwire/pgcode/codes.go#L383)):*
> ```Go
> 	InternalConnectionFailure = MakeCode("58C01")
> 
> 	// Class XCR - cockroach extension.
> ```

Classes are two characters. Make this "XC"
___
*[pkg/sql/pgwire/pgcode/codes.go, line 388 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/70347#-Mjo2Jw73rOE27mM2ZIX:-Mjo2Jw73rOE27mM2ZIY:b-35ncit) ([raw file](https://github.com/cockroachdb/cockroach/blob/59f84d2290bfcff5a39574ead151445355a134fc/pkg/sql/pgwire/pgcode/codes.go#L388)):*
> ```Go
> 	// UnsatisfiableBoundedStaleness signals that the bounded staleness query
> 	// cannot be satisfied.
> 	UnsatisfiableBoundedStaleness = MakeCode("XCR00")
> ```

"XCUBS"?


<!-- Sent from Reviewable.io -->

otan committed:
pgcode: use XC instead of CDB

Release note (sql change): Change the pgerror code XC instead of CD
for CockroachDB specific errors. This is because the "C" class is
reserved for the SQL standard. The pgcode `CDB00` used for
unsatisfiable bounded staleness is now `XCUBS`.

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/70347)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @rafiss)

___
*[pkg/sql/pgwire/pgcode/codes.go, line 383 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/70347#-Mjo2BRnEbnQnrZQXTxb:-Mjzqsdr6Kj5amYt6OVE:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/59f84d2290bfcff5a39574ead151445355a134fc/pkg/sql/pgwire/pgcode/codes.go#L383)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Classes are two characters. Make this "XC"
</blockquote></details>

Done.
___
*[pkg/sql/pgwire/pgcode/codes.go, line 388 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/70347#-Mjo2Jw73rOE27mM2ZIX:-MjzqsvaEwVNcFQKwohs:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/59f84d2290bfcff5a39574ead151445355a134fc/pkg/sql/pgwire/pgcode/codes.go#L388)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

"XCUBS"?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

knz reviewed (approved):
> Reviewed 1 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/70347)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @rafiss)



<!-- Sent from Reviewable.io -->

otan commented:
thanks for catching!

bors r=knz
